### DISCUSSION TITLE
Help Needed: Adding a Custom Network/Chain in reown/appkit/networks

### DISCUSSION THREAD

**Discussion by danishfik** [reown-com/appkit#2950] [Q&A] [answered]
Hi everyone,

I'm currently working on a project using Vue and ethers.js, and I need some help. I'm trying to add a new custom network/chain (PulseChain) to the appkit, but I haven't had any success so far.

I know the default appkit comes with networks like mainnet, arbitrum, and polygon from @reown/appkit/networks, but PulseChain isn't included. Here’s what I've tried so far:

```javascript
import { mainnet, arbitrum, polygon } from '@reown/appkit/networks'

const pulsechain = {
    name: 'PulseChain',
    chainId: 369,
    rpcUrls: ['https://rpc.pulsechain.com'],
    explorerUrl: 'https://scan.pulsechain.com',
    nativeCurrency: {
      name: 'Pulse',
      symbol: 'PLS',
      decimals: 18
    }
}

const appkit = createAppKit({
    adapters: [new EthersAdapter()],
    networks: [pulsechain],
    metadata,
    allWallets: 'SHOW',
    projectId,
    includeWalletIds: [
        'e7c4d26541a7fd84dbdfa9922d3ad21e936e13a7a0e44385d44f006139e44d3b',
        'c57ca95b47569778a828d19178114f4db188b89b763c899ba0be274e97267d96',
        '4622a2b2d6af1c9844944291e5e7351a6aa24cd7b23099efac1b2fd875da31a0',
        '8a0ee50d1f22f6651afcae7eb4253e52a3310b90af5daef78a8c4929a9bb99d4'
    ],
    features: {
      analytics: false,
      email: false,
      socials: [],
      emailShowWallets: false,
      swaps: false,
      onramp: false
    }
});

``` 

Unfortunately, it doesn't seem to work, and I’m not sure what I'm missing. Does anyone know if it's possible to add custom networks like this, or if I’m doing something wrong?

Any help or insights would be greatly appreciated!

Thanks in advance!

**Comment by hskang9** [comment]
Reown is not compatible with Wagmi not having Multicall3. I would rather use walletconnect's web3modal until they have next release.

**Reply by hskang9** [reply]
Reown started to follow CAIP standard but CAIP is made by wallet devs and over entitled people who just wants to have space in this crypto stuff. They never made real apps nor dealt with corner cases, so it would just make app devs life much harder.

**Comment by rtomas** [comment, marked as answer]
please use this code. We are working on improving this network package very soon.

```
const pulsechain = {
  id: 'eip155:369',
  name: 'PulseChain',
  currency: 'PLS',
  chainId: 369,
  rpcUrl: 'https://rpc.pulsechain.com',
  explorerUrl: 'https://scan.pulsechain.com',
  nativeCurrency: {
    name: 'Pulse',
    symbol: 'PLS',
    decimals: 18
  },
  chainNamespace: 'eip155'
} as const 
```
